diff -ruN org.nx.private.systemtray/contents/ui/ExpanderArrow.qml org.nx.private.systemtray/contents/ui/ExpanderArrow.qml
--- org.nx.private.systemtray/contents/ui/ExpanderArrow.qml
+++ org.nx.private.systemtray/contents/ui/ExpanderArrow.qml
@@ -70,13 +70,13 @@
             svg: arrowSvg
             elementId: {
                 if (plasmoid.location == PlasmaCore.Types.TopEdge) {
-                    return "down-arrow";
+                    return "right-arrow";
                 } else if (plasmoid.location == PlasmaCore.Types.LeftEdge) {
                     return "right-arrow";
                 } else if (plasmoid.location == PlasmaCore.Types.RightEdge) {
                     return "left-arrow";
                 } else {
-                    return "up-arrow";
+                    return "left-arrow";
                 }
             }
         }
@@ -102,13 +102,13 @@
             svg: arrowSvg
             elementId: {
                 if (plasmoid.location == PlasmaCore.Types.TopEdge) {
-                    return "up-arrow";
+                    return "left-arrow";
                 } else if (plasmoid.location == PlasmaCore.Types.LeftEdge) {
                     return "left-arrow";
                 } else if (plasmoid.location == PlasmaCore.Types.RightEdge) {
                     return "right-arrow";
                 } else {
-                    return "down-arrow";
+                    return "right-arrow";
                 }
             }
         }
diff -ruN org.nx.private.systemtray/contents/ui/main.qml org.nx.private.systemtray/contents/ui/main.qml
--- org.nx.private.systemtray/contents/ui/main.qml
+++ org.nx.private.systemtray/contents/ui/main.qml
@@ -19,6 +19,7 @@
 
 import QtQuick 2.5
 import QtQuick.Layouts 1.1
+import QtQuick.Window 2.2
 import org.kde.plasma.core 2.0 as PlasmaCore
 import org.kde.plasma.plasmoid 2.0
 import org.kde.draganddrop 2.0 as DnD
@@ -241,7 +242,7 @@
     CurrentItemHighLight {
         visualParent: tasksRow
         target: root.activeApplet && root.activeApplet.parent.parent == tasksRow ? root.activeApplet.parent : root
-        location: plasmoid.location
+        location: PlasmaCore.Types.RightEdge
     }
 
     DnD.DropArea {
@@ -336,9 +337,10 @@
     //Main popup
     PlasmaCore.Dialog {
         id: dialog
-        visualParent: root
+        x: Screen.desktopAvailableWidth - dialog.width
+        
         flags: Qt.WindowStaysOnTopHint
-        location: plasmoid.location
+        location: PlasmaCore.Types.RightEdge
         hideOnWindowDeactivate: expandedRepresentation.hideOnWindowDeactivate
 
         onVisibleChanged: {
