diff -ruN org.kde.plasma.private.systemtray/contents/ui/ExpandedRepresentation.qml org.kde.plasma.private.systemtray/contents/ui/ExpandedRepresentation.qml
--- org.kde.plasma.private.systemtray/contents/ui/ExpandedRepresentation.qml
+++ org.kde.plasma.private.systemtray/contents/ui/ExpandedRepresentation.qml
@@ -7,6 +7,7 @@
 
 import QtQuick 2.12
 import QtQuick.Layouts 1.12
+import QtQuick.Window 2.2
 
 import org.kde.plasma.core 2.0 as PlasmaCore
 // We still need PC2 here for that version of Menu, as PC2 Menu is still very problematic with QActions
@@ -19,7 +20,7 @@
     id: popup
     //set width/height to avoid useless Dialog resize
     readonly property int defaultWidth: PlasmaCore.Units.gridUnit * 24
-    readonly property int defaultHeight: PlasmaCore.Units.gridUnit * 24
+    readonly property int defaultHeight: Screen.desktopAvailableHeight
 
     width: defaultWidth
     Layout.minimumWidth: defaultWidth
diff -ruN org.kde.plasma.private.systemtray/contents/ui/ExpanderArrow.qml org.kde.plasma.private.systemtray/contents/ui/ExpanderArrow.qml
--- org.kde.plasma.private.systemtray/contents/ui/ExpanderArrow.qml
+++ org.kde.plasma.private.systemtray/contents/ui/ExpanderArrow.qml
@@ -54,13 +54,13 @@
             svg: arrowSvg
             elementId: {
                 if (plasmoid.location === PlasmaCore.Types.TopEdge) {
-                    return "down-arrow";
+                    return "right-arrow";
                 } else if (plasmoid.location === PlasmaCore.Types.LeftEdge) {
                     return "right-arrow";
                 } else if (plasmoid.location === PlasmaCore.Types.RightEdge) {
                     return "left-arrow";
                 } else {
-                    return "up-arrow";
+                    return "left-arrow";
                 }
             }
         }
@@ -86,13 +86,13 @@
             svg: arrowSvg
             elementId: {
                 if (plasmoid.location === PlasmaCore.Types.TopEdge) {
-                    return "up-arrow";
+                    return "left-arrow";
                 } else if (plasmoid.location === PlasmaCore.Types.LeftEdge) {
                     return "left-arrow";
                 } else if (plasmoid.location === PlasmaCore.Types.RightEdge) {
                     return "right-arrow";
                 } else {
-                    return "down-arrow";
+                    return "right-arrow";
                 }
             }
         }
diff -ruN org.kde.plasma.private.systemtray/contents/ui/main.qml org.kde.plasma.private.systemtray/contents/ui/main.qml
--- org.kde.plasma.private.systemtray/contents/ui/main.qml
+++ org.kde.plasma.private.systemtray/contents/ui/main.qml
@@ -7,6 +7,7 @@
 
 import QtQuick 2.5
 import QtQuick.Layouts 1.1
+import QtQuick.Window 2.2
 import org.kde.plasma.core 2.1 as PlasmaCore
 import org.kde.plasma.plasmoid 2.0
 import org.kde.draganddrop 2.0 as DnD
@@ -47,7 +48,7 @@
     }
 
     CurrentItemHighLight {
-        location: plasmoid.location
+        location: PlasmaCore.Types.LeftEdge
         parent: root
     }
 
@@ -205,9 +206,10 @@
     //Main popup
     PlasmaCore.Dialog {
         id: dialog
-        visualParent: root
+        x: Screen.desktopAvailableWidth
+        
         flags: Qt.WindowStaysOnTopHint
-        location: plasmoid.location
+        location: PlasmaCore.Types.LeftEdge
         hideOnWindowDeactivate: !plasmoid.configuration.pin
         visible: systemTrayState.expanded
         backgroundHints: (plasmoid.containmentDisplayHints & PlasmaCore.Types.DesktopFullyCovered) ? PlasmaCore.Dialog.SolidBackground : PlasmaCore.Dialog.StandardBackground
